### PR TITLE
Fix environment variable not accepting lists

### LINKED ISSUE #11618: PREFECT_TASK_DEFAULT_RETRY_DELAY_SECONDS environment variable does not accept a list
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

When passing a list to the environment variable PREFECT_TASK_DEFAULT_RETRY_DELAY_SECONDS the server is unable to start up.

### Reproduction

```python3
PREFECT_TASK_DEFAULT_RETRY_DELAY_SECONDS=[10, 20, 40]
```

### Error

```python3
Error loading ASGI app factory: unhashable type: 'list'
Server stopped!
```

### Versions

```Text
Version:             2.14.13
API version:         0.8.4
Python version:      3.11.7
Git commit:          9df5ba72
Built:               Thu, Jan 4, 2024 5:09 PM
OS/Arch:             linux/x86_64
Profile:             default
Server type:         ephemeral
Server:
  Database:          postgresql
```

### Additional context

https://github.com/PrefectHQ/prefect/blob/cbd8b8f0f80b505077da927094eb6f9b44f6ba30/src/prefect/settings.py#L755-L757

### PR DESCRIPTION
**Pull request by clefelhocz2**
This PR is one way to closes #11618.   The problem is that lists were not hashable which was described here: https://github.com/PrefectHQ/prefect/issues/11618#issuecomment-1887558096.   In scanning settings I didn't see any other settings which had this issue.  That being said, the note also points out that in the future this may change.  

In reviewing settings.py I noticed that environment variables are also cached, but using the text string representations to hash.  This is seen in get_setting_from_env.   Rather than going through and checking each variable and creating special hashing functions, it feels like converting to strings using to_environment_variables and hashing is a simpler solution.  

I also changed a test to verify the list actually flows through the code.

I haven't updated checklist or proceeded since this is my first PR in this repo and am waiting for feedback.  

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

File: src/prefect/server/api/server.py [hunk 1]
```diff
@@ -499,7 +499,9 @@ def create_app(
             and services will be disabled.
     """
     settings = settings or prefect.settings.get_current_settings()
-    cache_key = (settings, ephemeral)
+    env_variables = settings.to_environment_variables()
+    settings_key = hash(tuple((key, value) for key, value in env_variables.items()))
```
Comment (bunchesofdonald) [inline src/prefect/server/api/server.py:503]: This feels like something that maybe our settings should handle. What do you think about adding a method to the settings that's like:

```python
def hash_key(self) -> str:
    env_variables = self.to_environment_variables()
    return str(hash(tuple((key, value) for key, value in env_variables.items())))
```

And then we can call that here: `cache_key = (settings.hash_key(), ephemeral)`

**Review by bunchesofdonald — COMMENTED** [review]
Thanks for creating this PR @clefelhocz2 ! I think this approach makes sense. Ideally we'd have a test that's specific to this functionality and I think we should pull the settings specific hash generation logic into the settings itself.

File: src/prefect/server/api/server.py (hunk 1 above)
Comment (clefelhocz2) [reply src/prefect/server/api/server.py:503]: Thanks for the feedback, I'll try to get to this in the next 48 hours.

**Comment by clefelhocz2** [comment]
Okay updated.   I did this as two commits, but it could easily be squashed.

File: src/prefect/server/api/server.py (hunk 1 above)
Comment (clefelhocz2) [reply src/prefect/server/api/server.py]: Updated PR.

**Review by bunchesofdonald — APPROVED** [review]
This looks great, thank you so much!

**Pull request merged**